eclass/nodejs-pack.eclass

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: nodejs-pack.eclass
+# @MAINTAINER:
+# Fco. Javier Félix <web@inode64.com>
+# @AUTHOR:
+# Fco. Javier Félix <web@inode64.com>
+# @SUPPORTED_EAPIS: 8
+# @BLURB: An eclass for build NodeJS projects
+# @DESCRIPTION:
+# An eclass providing functions to build NodeJS packages
+#
+# Credits and ideas from:
+#   Initial version from:
+#       https://github.com/gentoo/gentoo/pull/930/files
+#       https://github.com/samuelbernardo/ssnb-overlay/blob/master/eclass/npm.eclass
+#       https://github.com/gentoo-mirror/lanodanOverlay/blob/master/eclass/nodejs.eclass
+#       https://github.com/Tatsh/tatsh-overlay/blob/master/eclass/yarn.eclass
+
+case ${EAPI} in
+8) ;;
+*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
+esac
+
+if [[ -z ${_NODEJS_PACK_ECLASS} ]]; then
+_NODEJS_PACK_ECLASS=1
+
+inherit nodejs
+
+if has nodejs-mod ${INHERITED}; then
+    eerror "nodejs-mod and nodejs-pack eclass are incompatible"
+fi
+
+RDEPEND+=" net-libs/nodejs"
+
+# Do not complain about CFLAGS etc since nodejs-pack projects do not use them.
+QA_FLAGS_IGNORED='.*'
+
+# Upstream does not support stripping go packages
+RESTRICT="test strip"
```

Comment:
Do we really always want to restrict tests? Why?

Reply:
I have added the test restriction because the production Nodejs-pack comes clean and eliminates the test codes and modules, I have seen several, none of them included the mods to perform the test.

Although it is true that it could be activated, but I believe in production I do not believe that support will be included.